lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -55,6 +55,37 @@
     private $replacementCharacters = array('__', '-');
 
     /**
+     * The mode that directories will be created with.
+     *
+     * @var int
+     */
+    protected $directoryMode = 0777;
+    
+    /**
+     * Cached objects are stored in directories.  These directory names are determined
+     * by splitting up the 32-char ID of the object into x parts.  If x=16 files will
+     * be two dirs deep (32/16=2).  If x=2 files will be eight dirs deep (32/2=16)
+     * 
+     * @var int
+     */
+    protected $directorySpreadChars = 2;
+
+    /**
+     * The mode that files will be created with.  Null means the file will be created
+     * with the current umask.
+     *
+     * @var int|null
+     */
+    protected $fileMode;
```

Comment:
this file mode is not used currently, and the directory mode is not used everywhere. I suggest dropping them (see my comment on the ``setDirectoryMode`` method for more details about my reasons)